examples/simple.rs

```diff
@@ -0,0 +1,263 @@
+    for line in layout.lines() {
+        // Iterate over GlyphRun's within each line
+        for glyph_run in line.glyph_runs() {
+            render_glyph_run(&mut scale_cx, &glyph_run, &mut img, padding);
+        }
+    }
+
+    // Write image to PNG file
+    let output_file = File::create("output.png").unwrap();
+    let png_encoder = PngEncoder::new(output_file);
+    img.write_with_encoder(png_encoder).unwrap();
+}
+
+fn render_glyph_run(
+    context: &mut ScaleContext,
+    glyph_run: &GlyphRun<Color>,
+    img: &mut RgbaImage,
+    padding: u32,
+) {
+    // Resolve properties of the GlyphRun
+    let mut run_x = glyph_run.offset();
+    let run_y = glyph_run.baseline();
+    let style = glyph_run.style();
+    let color = style.brush;
+
+    // Get the "Run" from the "GlyphRun"
+    let run = glyph_run.run();
+
+    // Resolve properties of the Run
+    let font = run.font();
+    let font_size = run.font_size();
+
+    // Convert from parley::Font to swash::FontRef
+    let font_ref = FontRef::from_index(font.data.as_ref(), font.index as usize).unwrap();
+
+    // Iterates over the glyphs in the GlyphRun
+    for glyph in glyph_run.glyphs() {
+        let glyph_id: GlyphId = glyph.id;
+        let glyph_x = run_x + glyph.x;
+        let glyph_y = run_y - glyph.y;
+        run_x += glyph.advance;
+        let Some(rendered_glyph) = render_glyph(
+            context,
+            &font_ref,
+            font_size,
+            true,
+            glyph_id,
+            glyph_x.fract(),
+            glyph_y.fract(),
+        ) else {
+            println!("No glyph");
+            continue;
+        };
+
+        let glyph_width = usize::try_from(rendered_glyph.placement.width).expect("usize < 32 bits");
+        let glyph_height =
+            usize::try_from(rendered_glyph.placement.height).expect("usize < 32 bits");
+        let glyph_origin_x =
+            glyph_x.floor() as i32 + rendered_glyph.placement.left + padding as i32;
+        let glyph_origin_y =
+            (glyph_y.floor() as i32) - rendered_glyph.placement.top + padding as i32;
+
+        match rendered_glyph.content {
+            Content::Mask => {
+                let mut i = 0;
+                for off_y in 0..glyph_height as i32 {
+                    for off_x in 0..glyph_width as i32 {
+                        let x = (glyph_origin_x + off_x) as u32;
+                        let y = (glyph_origin_y + off_y) as u32;
+                        let alpha = rendered_glyph.data[i];
+                        if alpha > 0 {
+                            let color = blend_px_naive(*img.get_pixel(x, y), color, alpha);
+                            img.put_pixel(x, y, color);
+                        }
+                        i += 1;
+                    }
+                }
+            }
+            Content::SubpixelMask => unimplemented!(),
+            Content::Color => {
+                for (off_y, row) in rendered_glyph.data.chunks_exact(glyph_width).enumerate() {
```

Comment:
Should be `glyph_width * 4` since we want the full stride for each row. The image will also need to be blended otherwise it’ll have a black background.

Reply:
Ah yes. With this change (+the same blending as I'm using for the alpha mask + #56 applied) emojis are rendering reasonably.

![output](https://github.com/linebender/parley/assets/1007307/37988ff5-1061-4f79-b9ea-bb837cab90ca)